javalib/src/main/scala/java/lang/String.scala

```diff
@@ -1,42 +1,951 @@
 package java.lang
 
 import java.io.Serializable
+import java.util.Comparator
+import java.util.Formatter
+import java.util.Locale
+import java.util.regex.Pattern
+import java.nio.ByteBuffer
+import java.nio.CharBuffer
+import java.nio.charset.Charset
+import java.nio.charset.IllegalCharsetNameException
+import java.nio.charset.UnsupportedCharsetException
+import java.security.AccessController
+import scala.util.Try
 
-class _String private() extends CharSequence with Serializable with Comparable[_String] {
-  def this(value: Array[scala.Char], offset: scala.Int, count: scala.Int) = this()
-
-  def length(): scala.Int = ???
-  def charAt(index: scala.Int): Char = ???
-  def compareTo(another_String: _String): scala.Int = ???
-  def subSequence(beginIndex: scala.Int, endIndex: scala.Int): CharSequence = ???
-  def concat(str: _String): _String = ???
-  def indexOf(ch: scala.Int): scala.Int = ???
-  def indexOf(ch: scala.Int, fromIndex: scala.Int): scala.Int = ???
-  def lastIndexOf(ch: scala.Int): scala.Int = ???
-  def lastIndexOf(ch: scala.Int, fromIndex: scala.Int): scala.Int = ???
-  def substring(beginIndex: scala.Int): _String = ???
-  def substring(beginIndex: scala.Int, endIndex: scala.Int): _String = ???
-  def startsWith(prefix: _String): scala.Boolean = ???
-  def toLowerCase(): _String = ???
-  def toCharArray(): Array[Char] = ???
-  def equalsIgnoreCase(str: _String): scala.Boolean = ???
-  def getChars(srcBegin: scala.Int, srcEnd: scala.Int, dst: Array[scala.Char],
-      dstBegin: scala.Int): Unit = ???
-  override def hashCode(): scala.Int = ???
-
-  // TODO: rest of the api
+class _String extends Serializable with Comparable[_String] with CharSequence {
+  private var value = new Array[Char](0)
+  private var offset = 0
+  private var count = 0
+  private override var hashCode: Int = _
```

Comment:
This one is not actually overriding the hashCode. Probably better use a different name for it like `cashedHashCode`.